server/gokbg3/grails-app/controllers/org/gokb/rest/UsersController.groovy

```diff
@@ -0,0 +1,189 @@
+package org.gokb.rest
+
+import grails.converters.JSON
+import grails.gorm.transactions.Transactional
+import grails.plugin.springsecurity.annotation.Secured
+import org.gokb.cred.User
+
+@Secured(['ROLE_USER', 'IS_AUTHENTICATED_FULLY'])
+@Transactional(readOnly = true)
+class UsersController {
+
+  static namespace = 'rest'
+
+  def userProfileService
+
+  @Secured(value = ['ROLE_ADMIN', 'IS_AUTHENTICATED_FULLY'], httpMethod = 'GET')
+  def show() {
+    def result = [data: [:]]
+      def user = User.get(params.id as int)
+      if (user) {
+        result.data = collectUserProps(user)
+      }
+    render result as JSON
+  }
+
+  @Secured(['ROLE_ADMIN', 'IS_AUTHENTICATED_FULLY'])
+  @Transactional
+  def search() {
+    def result = [data: []]
+    // parse params
+    int offset = params.offset ? params.offset as int : 0
+    int limit = params.limit ? params.limit as int : 10
+    String[] sortFields, sortOrders
+    if (params._sort)
+      sortFields = params._sort.split(',')
+    if (params._order)
```

Comment:
Maybe we want to assume `ASC` as default for sortOrders? So `_sort` would suffice for sorting to work.

Reply:
done